### PR TITLE
Skuba aws cloud provider

### PR DESCRIPTION
**Pull request by flavio**
## Why is this PR needed?

This PR adds AWS support to the cloud-provider feature of skuba.

## What does this PR do?

Users can enable the AWS CPI integration when creating a cluster by issuing the following command:

```
skuba cluster init --control-plane <EBL created by terraform> --cloud-provider aws my-cluster
```

## Anything else a reviewer needs to know?

This PR is built on top of https://github.com/SUSE/skuba/pull/637. The former PR introduces all the terraform related changes needed to deploy CaaS Platform on top of AWS. This one introduces the `skuba` related changes.

The PR introduces also unit testing for the `cluster init` command. Due to that some code had to be  moved across the files to make it reusable inside of the tests.

## Info for QA

Once the cluster is deployed all the nodes will have the AWS CPI enabled. This can be verified by executing the following command:

```
kubectl get nodes  | grep -v NAME | awk {'print $1'} | xargs -d "\n" -i bash -c "echo {} && kubectl describe node {} | grep -i provid"
```

All the nodes should have a `Provider-ID` key set to a string with `aws://` inside of it. 

### Related info

Info that can be relevant for QA:
* link to other PRs that should be merged together: https://github.com/SUSE/skuba/pull/637
* link to packages that should be released together: the skuba package is NOT going to have the terraform states related with AWS. This is intended. 
* upstream issues: none

### Status **BEFORE** applying the patch

The nodes do not have the AWS CPI enabled. Nothing will be returned by:

```
kubectl get nodes  | grep -v NAME | awk {'print $1'} | xargs -d "\n" -i bash -c "echo {} && kubectl describe node {} | grep -i provid"
```

### Status **AFTER** applying the patch

The nodes will have the AWS CPI enabled. This can be verified with the instructions written at the beginning of this section.

## Docs

The official docs do not have a AWS section published. We will enable that once we agree to make this feature supproted.

### CODE REVIEW

File: pkg/skuba/actions/cluster/init/constants.go [hunk 1]
```diff
@@ -62,5 +62,6 @@ var (
 				Content:  openstackReadme,
 			},
 		},
+		"aws": {},
```
Comment (ereslibre) [inline pkg/skuba/actions/cluster/init/constants.go]: Do you think we could at least include a `README.md` file inside the directory?

Comment (Klaven) [reply pkg/skuba/actions/cluster/init/constants.go]: I second the `README.md` this way we can say in the docs "read the readme for you platform" this way documentation can be consistent and handle this with a readme.

File: cmd/skuba/cluster/init.go [hunk 2]
```diff
@@ -46,30 +40,17 @@ func NewInitCmd() *cobra.Command {
...
-				ClusterName:       args[0],
-				CloudProvider:     initOptions.CloudProvider,
-				ControlPlane:      initOptions.ControlPlane,
-				PauseImage:        images.GetGenericImage(skuba.ImageRepository, "pause", kubernetes.ComponentVersionForClusterVersion(kubernetes.Pause, kubernetesVersion)),
-				KubernetesVersion: kubernetesVersion,
-				ImageRepository:   skuba.ImageRepository,
-				EtcdImageTag:      kubernetes.ComponentVersionForClusterVersion(kubernetes.Etcd, kubernetesVersion),
-				CoreDNSImageTag:   kubernetes.ComponentVersionForClusterVersion(kubernetes.CoreDNS, kubernetesVersion),
-				StrictCapDefaults: initOptions.StrictCapDefaults,
+			initConfig, err := cluster.NewInitConfiguration(
```
Comment (Klaven) [inline cmd/skuba/cluster/init.go]: nice cleanup.

File: pkg/skuba/actions/cluster/init/init_test.go [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+			t.Errorf("Error while inspecting file %s: %v", f, err)
+		}
+	}
+}
+
+// check the init configuration hold inside of `file`. Path to file is built starting
+// from the test `ctx` and the `clusterName`, plus the ending name of the `file`.
+// `cloud` holds the name of the CPI - leave empty if CPI is disabled.
+func checkInitConfig(ctx TestFilesystemContext, clusterName, file, cloud string) error {
+	expected := cloud != ""
```
Comment (Klaven) [inline pkg/skuba/actions/cluster/init/init_test.go]: nit: `expected := len(cloud) > 0` would cover `nil`

**Comment by rssfed23** [comment]
When I run this I get the following (using an admin access key)
```
Error: Error applying plan:

4 error(s) occurred:

* aws_instance.nodes[0]: 1 error(s) occurred:

* aws_instance.nodes.0: Error launching source instance: InvalidParameterValue: Value (caasp-k8s-worker-vm-profile) for parameter iamInstanceProfile.name is invalid. Invalid IAM Instance Profile name
        status code: 400, request id: f955d665-924a-476b-9728-16a7457ace78
* aws_instance.control_plane: 1 error(s) occurred:

* aws_instance.control_plane: Error launching source instance: InvalidParameterValue: Value (caasp-k8s-master-vm-profile) for parameter iamInstanceProfile.name is invalid. Invalid IAM Instance Profile name
        status code: 400, request id: 70748c79-7874-4575-93ef-189fad1241c2
* aws_instance.nodes[1]: 1 error(s) occurred:

* aws_instance.nodes.1: Error launching source instance: InvalidParameterValue: Value (caasp-k8s-worker-vm-profile) for parameter iamInstanceProfile.name is invalid. Invalid IAM Instance Profile name
        status code: 400, request id: 6a0199c0-655d-41b4-a8c0-f4d0868bc757
* aws_instance.nodes[2]: 1 error(s) occurred:

* aws_instance.nodes.2: Error launching source instance: InvalidParameterValue: Value (caasp-k8s-worker-vm-profile) for parameter iamInstanceProfile.name is invalid. Invalid IAM Instance Profile name
        status code: 400, request id: df8f22c8-8ee2-4dc1-a0bb-98e67665f50e```

File: pkg/skuba/actions/cluster/init/init_test.go (hunk 3 above)
Comment (flavio) [reply pkg/skuba/actions/cluster/init/init_test.go]: changed

**Comment by flavio** [comment]
Once everything is fine I can squash all the commits into a single one

**Comment by flavio** [comment]
> PS. Love that the tests come with the code +1

@Klaven once this gets merged I'll wait for you/someone else to provide test coverage for the openstack CPI :stuck_out_tongue:

**Pull request merged**